src/index.rs

```diff
@@ -0,0 +1,18 @@
+// We need this trait to allow panics when the index is out of range.
```

Comment:
Should be documented (even if it's just a short "this is the index trait (auto implemented by usize)...").